### PR TITLE
chore: use `composeRef(ref, internalRef)` replace `ref || internalRef`

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | chore: use `composeRef(ref, internalRef)` replace `ref || internalRef` |
| 🇨🇳 Chinese | 代码优化，无需纳入 ChangeLog |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at f3e0742</samp>

Refactor `Button` component to use named React imports, `composeRef`, and fix TypeScript errors. Optimize `fixTwoCNChar` function.

### <samp>🤖 Generated by Copilot at f3e0742</samp>

*  Use named imports for React and its hooks and utilities, and add `composeRef` utility to combine forwarded and internal refs ([link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L4-R13), [link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L113-R145), [link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L174-R178), [link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L274-R281))
*  Change ref types to `any` or `React.Ref<HTMLElement>` to avoid TypeScript errors ([link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L113-R145), [link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L174-R178), [link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L240-R253), [link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L260-R267))
*  Rename `isNeedInserted` to `needInserted` and make it a constant instead of a function ([link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L113-R145), [link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L174-R178), [link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L240-R253))
*  Move `fixTwoCNChar` function to a `useEffect` hook to avoid unnecessary calls and dependencies ([link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L174-R178))
*  Use `useMemo` instead of `React.useMemo` to follow the named imports convention ([link](https://github.com/ant-design/ant-design/pull/42036/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L113-R145))

### CODE REVIEW

File: components/button/button.tsx [hunk 1]
```diff
@@ -237,41 +241,44 @@ const InternalButton: React.ForwardRefRenderFunction<
...
     >
       {iconNode}
       {kids}
     </button>
   );
 
-  if (!isUnBorderedButtonType(type)) {
-    buttonNode = <Wave disabled={!!innerLoading}>{buttonNode}</Wave>;
+  if (isUnBorderedButtonType(type)) {
+    return wrapSSR(buttonNode);
```
Comment (MadCcc) [inline components/button/button.tsx:275]: 这个不如原来的写法，wrapSSR 用了两遍

Comment (li-jia-nan) [reply components/button/button.tsx]: done~

**Review by MadCcc — APPROVED** [review]

**Comment by afc163** [comment]
composeRef 有啥能力？

**Comment by li-jia-nan** [comment]
> composeRef 有啥能力？

将多个 ref 合并到一个 ref 函数里面：

```ts
function fillRef<T>(ref: React.Ref<T>, node: T) {
  if (typeof ref === 'function') {
    ref(node);
  } else if (typeof ref === 'object' && ref && 'current' in ref) {
    (ref as any).current = node;
  }
}

/**
 * Merge refs into one ref function to support ref passing.
 */
function composeRef<T>(...refs: React.Ref<T>[]): React.Ref<T> {
  const refList = refs.filter(ref => ref);
  if (refList.length <= 1) {
    return refList[0];
  }
  return (node: T) => {
    refs.forEach(ref => {
      fillRef(ref, node);
    });
  };
}
```

**Pull request merged**